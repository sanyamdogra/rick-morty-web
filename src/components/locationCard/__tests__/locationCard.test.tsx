import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import LocationCard from "..";

afterEach(cleanup);

test("should match the snapshot and render correct data", () => {
  const { asFragment, rerender } = render(
    <LocationCard
      loading={false}
      locationName={"Test Location Name"}
      dimension={"Test Location Dimension"}
      noOfResidents={7}
      type={"Test Location Type"}
    />
  );

  expect(asFragment()).toMatchSnapshot();

  expect(screen.getByTestId("location-card-heading")).toHaveTextContent(
    "Test Location Name"
  );

  expect(screen.queryByTestId("loader-wrapper")).toBeNull();

  rerender(
    <LocationCard
      loading={true}
      locationName={"Test Location Name"}
      dimension={"Test Location Dimension"}
      noOfResidents={7}
      type={"Test Location Type"}
    />
  );

  expect(screen.getByTestId("loader-wrapper")).toHaveTextContent(
    "Fetching location info ..."
  );

  expect(screen.queryByTestId("card-field-label")).toBeNull();
});
