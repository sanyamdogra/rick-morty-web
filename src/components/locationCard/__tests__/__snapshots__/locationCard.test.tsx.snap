// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot and render correct data 1`] = `
<DocumentFragment>
  <div
    class="rick-morty-loc-card"
  >
    <div
      class="rick-morty-loc-card__heading"
      data-testid="location-card-heading"
    >
      Test Location Name
    </div>
    <div
      class="rick-morty-card-field__label"
      data-testid="card-field-label"
    >
      Dimension
    </div>
    <div
      class="rick-morty-card-field__value"
      data-testid="card-field-value"
    >
      Test Location Dimension
    </div>
    <div
      class="rick-morty-card-field__label"
      data-testid="card-field-label"
    >
      No of residents
    </div>
    <div
      class="rick-morty-card-field__value"
      data-testid="card-field-value"
    >
      7
    </div>
    <div
      class="rick-morty-card-field__label"
      data-testid="card-field-label"
    >
      Type
    </div>
    <div
      class="rick-morty-card-field__value"
      data-testid="card-field-value"
    >
      Test Location Type
    </div>
  </div>
</DocumentFragment>
`;
